crates/sui-graphql-rpc/src/types/move_registry/named_type.rs

```diff
@@ -55,7 +56,13 @@ impl NamedType {
 
         let correct_type_tag: String = Self::replace_names(name, &name_package_id_mapping)?;
 
-        TypeTag::from_str(&correct_type_tag).map_err(|e| Error::Client(format!("bad type: {e}")))
+        let tag = TypeTag::from_str(&correct_type_tag)
+            .map_err(|e| Error::Client(format!("bad type: {e}")))?;
+
+        resolver
+            .canonical_type(tag)
+            .await
+            .map_err(|e| Error::Client(format!("Failed to retrieve type: {e}")))
```

Comment:
nit: Should this be an internal error? I can't think of a way for the client to mess this step up if all the previous steps have succeeded.